### ISSUE TITLE
Request: Add a way to round inside shape border.

### ISSUE THREAD

**Issue by bernaferrari** [flutter/flutter#104201]
By default, Flutter squares the inner border of RoundedRectangleBorder (and BoxDecoration).

<img width="646" alt="image" src="https://user-images.githubusercontent.com/351125/169363557-2208bb0e-40f6-4f46-a133-cde7731e6644.png">

Figma doesn't do this, and with the new StrokeAlign, I think things would be more consistent with this feel:
<img width="768" alt="image" src="https://user-images.githubusercontent.com/351125/169363800-6e30605d-41b1-43a1-9726-90d5c27a78df.png">

**Comment by gspencergoog** [comment]
It seems to me that you'd actually want to reduce the inner radius based on how far inside the shape the border is, with a lower limit of zero, instead of using the same radius as the outer radius.

**Comment by bernaferrari** [comment]
I would expect, coming from design tools,  for the inner radius to be circular. Flutter gets this right for smaller values, but on larger values it 'boxes' it. So you may say Flutter is kind of inconsistent because the behavior changes depending on the value.

I think one day someone will invent this and Flutter won't be able to make it:

<img width="340" alt="image" src="https://user-images.githubusercontent.com/351125/170546922-9cb964a1-6d5d-4dab-9c41-4fc77e4c47e0.png">

**Comment by HansMuller** [comment]
This can get complicated: 
http://hansmuller-webkit.blogspot.com/2013/02/padding-rounded-rectangle.html

The original demo seems to have disappeared although there's a version of polygonal shapes here: https://codepen.io/HansMuller/pen/AgLWaz

**Comment by bernaferrari** [comment]
I agree it can get complicated, but I think as I user I would expect a growing border to keep the same shape, not become rectangle, because design tools do that. Yes, there would be slightly clipping, but I prefer that. The user can add more padding. Or, if they don't like the circle, they can add a Container to the Text so it changes the background color. It is simple. The other way, a rounded internal is really hard to reproduce, but the internal implementation is kind of a single line of code, so I think the tradeoff could be worthy. 

<img width="358" alt="image" src="https://user-images.githubusercontent.com/351125/170560752-5a4ea02a-3d04-4cbd-aa94-1cf77e242e3e.png">

**Comment by HansMuller** [comment]
Expanding a rectangle to a rounded rectangle but shrinking a rounded rectangle to a smaller rounded rectangle is asymmetrical; seems odd to have the original shape disappear if you grow the border and then shrink it back by the same amount.

**Comment by bernaferrari** [comment]
Let's forget that idea then. This is what is weird to me, there is some kind of radius right now, but it doesn't match what you would expect. Until 30, it is square, then it starts becoming rounded. I wish it was always rounded:

https://user-images.githubusercontent.com/351125/170574449-4d5036dc-d63c-4c2a-91f0-4839dfd0a4dc.mov
